### PR TITLE
feature(ios): Add processor count to device info

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] New feature

## :scroll: Description

Add CPU count from iOS. Android works already since the version it was added to `sentry-java` no changes in RN are needed.

## :bulb: Motivation and Context

- closes: https://github.com/getsentry/sentry-react-native/issues/2884

## :green_heart: How did you test it?
sample app

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: ios/RNSentry.mm [hunk 1]
```diff
@@ -207,6 +207,19 @@ - (void)setEventEnvironmentTag:(SentryEvent *)event
         }
     }];
 
+    NSDictionary<NSString *, id> *extraContext = [PrivateSentrySDKOnly getExtraContext];
```
Comment (marandaneto) [inline ios/RNSentry.mm:210]: Is that a new method? we don't have that on Flutter, wondering if we should.

File: ios/RNSentry.mm [hunk 2]
```diff
@@ -207,6 +207,19 @@ - (void)setEventEnvironmentTag:(SentryEvent *)event
...
+
+    NSMutableDictionary<NSString *, NSDictionary<NSString *, id> *> *deviceContext = [contexts[@"context"][@"device"] mutableCopy];
+    [deviceContext addEntriesFromDictionary:extraContext[@"device"]];
```
Comment (marandaneto) [inline ios/RNSentry.mm:213]: M: What if `extraContext` is null? Or what if `extraContext` does not have `device`, does ObjC handle that automatically or would that throw?

File: ios/RNSentry.mm (hunk 1 above)
Comment (krystofwoldrich) [reply ios/RNSentry.mm:210]: Yes, it's a new method, [added here](https://github.com/getsentry/sentry-cocoa/pull/2855/files#diff-127399cc36e15a55aa8f32df346b6407d4e9faab0cf65b94188cbd8ba2c9bbed).

I believe flutter should have it too.

File: ios/RNSentry.mm (hunk 2 above)
Comment (krystofwoldrich) [reply ios/RNSentry.mm:213]: Currently, these two keys are always there but I agree it's not safe.

Comment (marandaneto) [reply ios/RNSentry.mm:213]: Check https://github.com/getsentry/sentry-dart/pull/1402
I always check if there's a value and it's the type I expect.
https://en.wikipedia.org/wiki/Defensive_programming

**Review by marandaneto — APPROVED** [review]

**Pull request merged**